shared/middleware/helpers/table.rb

```diff
@@ -69,6 +74,8 @@ def update(id, value, ip_address)
   end
   
   def to_a()
+    raise NotFound unless @can_enumerate
```

Comment:
Same suggestion as with to_hash, `return [] unless @can_enumerate` is gets the same job done with less complexity for the consumer.